draft-wussler-openpgp-forwarding.md

```diff
@@ -168,39 +178,66 @@ keys and computing separate proxy transformation parameters.
 
 ## Key Flag 0x40 {#flag-forwarding}
 
-The flag 0x40 is added to signal "This key may be used for forwarded communication",
-this is to be used on subkeys for decryption of forwarded messages, i.e. forwardee subkeys.
+The key flag 0x40 is added to the first octet of the key flags (Table 12 of {{I-D.ietf-openpgp-crypto-refresh}}).
+It indicates that the key may be used to decrypt forwarded communications.
 
-This is designed to distinguish the usage from the existing 0x04 flag,
-preventing implementations not capable of forwarding from using this key for
+This is intended to prevent implementations unaware of forwarding keys from using this key for
 direct encryption, and thus generating unreadable messages.
 
-An implementation SHOULD NOT export public subkeys flagged as 0x40.
-A public key directory SHOULD NOT accept subkeys flagged as 0x40.
+An implementation SHOULD NOT export public subkeys with key flag 0x40.
+A public key directory SHOULD NOT accept subkeys with key flag 0x40.
 
-Keys having this flag MUST have the forwarding KDF parameters version 0xFF
+Keys with this flag MUST have the forwarding KDF parameters version 0xFF
 defined in {{generating-forwarding-key}}.
 
 # Setting up a forwarding instance
 
-Starting from an OpenPGP v4 certificate as defined in {{I-D.ietf-openpgp-crypto-refresh}} with a
-Curve25519 encryption-only subkey in this section is described how to compute a
-proxy transformation parameter and a forwardee subkey.
+This section describes how to compute a proxy transformation parameter and a forwardee subkey
+for a v4 OpenPGP certificate with a Curve25519 encryption-only subkey.
 
-The original key MUST have an ECDH (Algorithm ID 18) as defined in {{I-D.ietf-openpgp-crypto-refresh}}
-Section 9.1. subkey with exclusively the 0x04 (encrypt communications) or 0x08 (encrypt storage) flags,
-as defined in {{I-D.ietf-openpgp-crypto-refresh}} Section 5.2.3.26.
+The original key MUST have an ECDH subkey (algorithm ID 18, as defined in Section 9.1 of
```

Comment:
It's a bit confusing to prescribe what a suitable key MUST be, either say what a suitable key is or what the key MUST be (and then it's implied that it's not suitable otherwise :)), e.g.
```suggestion
The subkeys used for forwarding MUST be ECDH keys (algorithm ID 18, as defined in Section 9.1 of
```